python-package/lightgbm/sklearn.py

```diff
@@ -27,11 +41,11 @@ def _objective_function_wrapper(func):
     ----------
     func: callable
         Expects a callable with signature ``func(y_true, y_pred)`` or ``func(y_true, y_pred, group):
-            y_true: array_like of shape [n_samples]
+            y_true: array-like of shape = [n_samples]
                 The target values
-            y_pred: array_like of shape [n_samples] or shape[n_samples * n_class] (for multi-class)
+            y_pred: array-like of shape = [n_samples] or shape = [n_samples * n_class] (for multi-class)
```

Comment:
I suppose, there should be "... or shape = [n_samples * **n_outputs**] (for **multi-output problem**)".

Reply:
@StrikerRUS we don't support multi-output yet.

Reply:
@wxchan 
Thanks for your review. Can you please explain than the case of `y` with `shape = [n_samples * n_class]`?
It's still 1d-array, as I understand and possible only in classification problem situation, right?

Reply:
@StrikerRUS yes, it's the flatten (n_samples, n_class) array.

Reply:
@wxchan 
Thank you. I'll edit `check_consistence_length` function with next commit.

And what about docstring (my first comment in this PR)? Do you know how it could be done?

Reply:
@wxchan 
Am I right that `fit` method accepts only `y` of shape = [n_samples] in both classification and regression case?

Reply:
@StrikerRUS yes for shape of y. I don't know how to do the docstring thing, you can simply add those to LGBMModel.__doc__ and add `Classifier only`.

Reply:
@wxchan
Thank you. I combined both ways of improving docstrings: dynamic construction and notes about classification problem only.